base/reflection.jl

```diff
@@ -1505,7 +1505,17 @@ See also: [`parentmodule`](@ref), and `@which` and `@edit` in [`InteractiveUtils
 function which(@nospecialize(f), @nospecialize(t))
     t = to_tuple_type(t)
     tt = signature_type(f, t)
-    return which(tt)
+    world = get_world_counter()
+    match, _ = Core.Compiler._findsup(tt, nothing, world)
+    if match === nothing
+        me = MethodError(f, t, world)
+        ee = ErrorException(sprint(io -> begin
+            println(io, "invoke() with these arguments would result in:");
```

Comment:
We talked about incorporating part of this text into the showerror itself, since it has the information to distinguish between an invoke and a call, but just calls it all "no method matches". What was the result of that attempt?

Reply:
We still were looking at updating the "no method matching " part of the text, to distinguish between call and invoke. It could say "no method defined for [call|invoke] to ", for example, if that fits the paragraph context